### PR TITLE
CI: Quadlet rootfs test: use container image as rootfs

### PR DESCRIPTION
**Pull request by edsantiago**
Test was written to use / (root). This is not parallel-safe.

Fixes: #23909

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

**Review by ygalblum — APPROVED** [review]
LGTM

File: test/system/252-quadlet.bats [hunk 1]
```diff
@@ -723,10 +723,18 @@ EOF
 @test "quadlet - rootfs" {
     skip_if_no_selinux
     skip_if_rootless
```
Comment (Luap99) [inline test/system/252-quadlet.bats:725]: I am somewhat sure these can be removed now?

Comment (edsantiago) [reply test/system/252-quadlet.bats:725]: Wouldn't the mounts need complicated `unshare`? Would those work inside quadlet?

Comment (Luap99) [reply test/system/252-quadlet.bats:725]: ah right yes you would need to use unshare rootless but yes as they all share the same userns the mounts should work with quadlet. Anyway I guess it is not really important to test it rootless. (although I do not get the no_selinux skip either)

I guess I just merge it as it is the same as before

/lgtm

**Pull request merged**